modules/custom/openy_upgrade_tool/src/OpenyUpgradeLogManager.php

```diff
@@ -0,0 +1,588 @@
+        $this->configExists = $entity;
+      }
+    }
+    else {
+      $config_name = $name;
+      $config = $this->configFactory->get($config_name);
+      $this->configExists = !$config->isNew() ? $config : FALSE;
+    }
+
+    // Use ConfigImporter validation.
+    $source_storage = new StorageReplaceDataWrapper($this->configStorage);
+    $source_storage->replaceData($config_name, $data);
+    $storage_comparer = new StorageComparer(
+      $source_storage,
+      $this->configStorage,
+      $this->configManager
+    );
+
+    if (!$storage_comparer->createChangelist()->hasChanges()) {
+      drupal_set_message($this->t('There are no changes to import.'), 'status');
+      return FALSE;
+    }
+    else {
+      $config_importer = new ConfigImporter(
+        $storage_comparer,
+        $this->eventDispatcher,
+        $this->configManager,
+        $this->lock,
+        $this->typedConfigManager,
+        $this->moduleHandler,
+        $this->moduleInstaller,
+        $this->themeHandler,
+        $this->getStringTranslation()
+      );
+
+      try {
+        $config_importer->validate();
+        $this->configImporter = $config_importer;
+      }
+      catch (ConfigImporterException $e) {
+        // There are validation errors.
+        $item_list = [
+          '#theme' => 'item_list',
+          '#items' => $config_importer->getErrors(),
+          '#title' => $this->t('The configuration cannot be imported because it failed validation for the following reasons:'),
+        ];
+        drupal_set_message($this->renderer->render($item_list));
+        return FALSE;
+      }
+    }
+
+    return TRUE;
+  }
+
+  /**
+   * Set status TRUE to OpenyUpgradeLog item (Conflict resolved).
+   *
+   * @param string $openy_upgrade_log
+   *   OpenyUpgradeLog item name.
+   * @param array $context
+   *   The batch context.
+   */
+  public static function setConflictResolvedBatchProcess($openy_upgrade_log, array &$context) {
+    $entity = \Drupal::service('openy_upgrade_log.manager')
+      ->loadByName($openy_upgrade_log);
+    $entity->setStatus(TRUE)->save();
+  }
+
+  /**
+   * Delete OpenyUpgradeLog item (no need to track customizations).
+   *
+   * @param string $openy_upgrade_log
+   *   OpenyUpgradeLog item name.
+   * @param array $context
+   *   The batch context.
+   */
+  public static function deleteOpenyUpgradeLogItemBatchProcess($openy_upgrade_log, array &$context) {
+    $entity = \Drupal::service('openy_upgrade_log.manager')
+      ->loadByName($openy_upgrade_log);
+    $entity->delete();
+  }
+
+  /**
+   * Finish batch.
+   *
+   * This function is a static function to avoid serializing the ConfigSync
+   * object unnecessarily.
+   */
+  public static function finishBatch($success, $results, $operations) {
+    if ($success) {
+      if (!empty($results['errors'])) {
+        foreach ($results['errors'] as $error) {
+          drupal_set_message($error, 'error');
+          \Drupal::logger('config_sync')->error($error);
```

Comment:
DI please with predefined logger channel

Reply:
We can't use DI in static methods.